### ISSUE TITLE
Accessibility: CheckboxButtonGroup should have a fieldset/legend structure.

### ISSUE THREAD

**Issue by oasisrising** [dohomi/react-hook-form-mui#336]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

![Screenshot 2024-12-17 at 11 06 39 AM](https://github.com/user-attachments/assets/fd69928c-2cb4-40cc-bb77-aa8e8e26da06)
Grouped controls should be part of a fieldset:
https://www.w3.org/WAI/tutorials/forms/grouping/#checkboxes

### Expected behavior 🤔

![Screenshot 2024-12-17 at 11 08 18 AM](https://github.com/user-attachments/assets/bbb262a7-f8e8-4690-b90f-f2f0ce9d6ed7)
Example with fieldset and legend components set on the MUI components:
https://mui.com/material-ui/react-checkbox/#formgroup

### Steps to reproduce 🕹

Steps:

1.
2.
3.
4.
